coderd/database/dbauthz/dbauthz.go

```diff
@@ -1474,13 +1474,12 @@ func (q *querier) GetUsersByIDs(ctx context.Context, ids []uuid.UUID) ([]databas
 	return q.db.GetUsersByIDs(ctx, ids)
 }
 
-// GetWorkspaceAgentByAuthToken is used in http middleware to get the workspace agent.
-// This should only be used by a system user in that middleware.
-func (q *querier) GetWorkspaceAgentByAuthToken(ctx context.Context, authToken uuid.UUID) (database.WorkspaceAgent, error) {
+func (q *querier) GetWorkspaceAgentAndOwnerByAuthToken(ctx context.Context, agentID uuid.UUID) (database.GetWorkspaceAgentAndOwnerByAuthTokenRow, error) {
```

Comment:
nit: ByAuthToken -> ByAgentId?

Reply:
This query selects by `workspace_agents.auth_token`.

Reply:
but requires `agentID uuid.UUID`, right?

Reply:
Ah that's a good point. The argument should be named `authToken uuid.UUID` because the auth token is also a UUID. Will change this 👍